iterations_runners/iterations_runner.java

```diff
@@ -188,32 +212,26 @@ public static void main(String args[]) throws
          *   - We print the result using toUnsignedString, thus interpreting
          *     the number as unsigned.
          */
-        long[] cycleCounts = new long[iterations];
-        Arrays.fill(cycleCounts, 0);
-
-        long[] aperfCounts = new long[iterations];
-        Arrays.fill(aperfCounts, 0);
-
-        long[] mperfCounts = new long[iterations];
-        Arrays.fill(mperfCounts, 0);
+        long[][] cycleCounts = new long[numCores][iterations];
+        long[][] aperfCounts = new long[numCores][iterations];
+        long[][] mperfCounts = new long[numCores][iterations];
+
+        /* longs default to zero, but we explicitely zero to force allocation */
+        for (int core = 0; core < numCores; core++) {
+            Arrays.fill(cycleCounts[core], 0);
+            Arrays.fill(mperfCounts[core], 0);
```

Comment:
This is a rather anal comment, but in other places you have `aperf` after `cycles` and `mperf` last; to avoid copy and paste errors it might be better to be consistent about that.